packages/material-ui/src/styles/experimentalStyled.js

```diff
@@ -64,34 +64,54 @@ const shouldForwardProp = (prop) => prop !== 'styleProps' && prop !== 'theme' &&
 const experimentalStyled = (tag, options, muiOptions = {}) => {
   const name = muiOptions.muiName;
   const defaultStyledResolver = styled(tag, { shouldForwardProp, label: name, ...options });
-  const muiStyledResolver = (...styles) => {
-    const stylesWithDefaultTheme = styles.map((stylesArg) => {
-      return typeof stylesArg === 'function'
-        ? ({ theme: themeInput, ...rest }) =>
-            stylesArg({ theme: isEmpty(themeInput) ? defaultTheme : themeInput, ...rest })
-        : stylesArg;
-    });
+  const muiStyledResolver = (styleArg, ...expressions) => {
+    const expressionsWithDefaultTheme = expressions
+      ? expressions.map((stylesArg) => {
+          return typeof stylesArg === 'function'
+            ? ({ theme: themeInput, ...rest }) => {
+                return stylesArg({
+                  theme: isEmpty(themeInput) ? defaultTheme : themeInput,
+                  ...rest,
+                });
+              }
+            : stylesArg;
+        })
+      : [];
+
+    let transformedStyleArg = styleArg;
+
+    if (Array.isArray(styleArg)) {
+      // If the type is array, than we need to add placeholders in the template for the overrides, variants and the sx styles
+      transformedStyleArg = [...styleArg, '', '', ''];
```

Comment:
This was my main point of interest since transpiling this results in quite a bit of babel helpers if array-spread isn't supported. But that's only the case for our /legacy bundle in v5 :tada: 

Though my the micro-perf part of my brain does want to dive into some v8 internals and see what's better :grin: 
```diff
-[...styleArg, '']
+styleArg.concat('')
```

Reply:
In my mind, when I see

```
[...styleArg, '']
```

I am able to immediately conclude that this is a copy of the array with additional elements, for every other syntax I would need to take a closer look to decide whether it's a copy or mutation :) Of course this is probably just my preference, but you are probably right that it requires more advanced transpiling behind the scene :D